routing/localchans/manager.go

```diff
@@ -71,17 +77,39 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 
 		// If we have a channel filter, and this channel isn't a part
 		// of it, then we'll skip it.
-		_, ok := chansToUpdate[info.ChannelPoint]
+		_, ok := unprocessedChans[info.ChannelPoint]
 		if !ok && haveChanFilter {
 			return nil
 		}
 
+		// Mark this channel as found by removing it. unprocessedChans
+		// will be used to report invalid channels later on.
+		delete(unprocessedChans, info.ChannelPoint)
+
 		// Apply the new policy to the edge.
 		err := r.updateEdge(info.ChannelPoint, edge, newSchema)
 		if err != nil {
-			log.Warnf("Cannot update policy for %v: %v\n",
-				info.ChannelPoint, err,
+			appendUpdateFailureItem(
```

Comment:
I've got a slight preference for having an `makeFailureItem` helper function then just going `failureUpdates = append(...` inline, I don't think it makes it less readable.

Reply:
Sounds good